mlflow/transformers/__init__.py

```diff
@@ -508,6 +543,18 @@ def save_model(
                 built_pipeline, input_example, model_config or inference_config
             )
 
+        # if pipeline is text-generation and a prompt template is specified,
+        # provide the return_full_text=False config by default to avoid confusing
+        # extra text for end-users
+        if prompt_template is not None and built_pipeline.task == "text-generation":
+            return_full_text_key = "return_full_text"
+            if model_config is None:
+                model_config = {return_full_text_key: False}
+                _logger.info(_PROMPT_TEMPLATE_RETURN_FULL_TEXT_INFO)
+            elif return_full_text_key not in model_config:
+                model_config[return_full_text_key] = False
+                _logger.info(_PROMPT_TEMPLATE_RETURN_FULL_TEXT_INFO)
```

Comment:
/ai is there a nice way to not duplicate the log call

Reply:
You could use 
```
model_config = model_config or {}
if return_full_text_key not in model_config:
...
```